File: frontend/src/utils/outfitItemCardText.ts
export interface OutfitItemCardText {
  shortName: string;
  oneLineReason: string | null;
}

/**
 * Split a long item description into a short name and optional one-line reason.
 */
export function parseOutfitItemCardText(description: string): OutfitItemCardText {
  const trimmed = description.trim();
  if (!trimmed) {
    return { shortName: '', oneLineReason: null };
  }

  const separators = [' — ', ' - ', ', ', '. '];
  for (const sep of separators) {
    const idx = trimmed.indexOf(sep);
    if (idx > 0 && idx < 60) {
      const shortName = trimmed.slice(0, idx).trim();
      const rest = trimmed.slice(idx + sep.length).trim();
      return {
        shortName,
        oneLineReason: rest.length > 0 ? rest.slice(0, 80) : null,
      };
    }
  }

  if (trimmed.length > 48) {
    return {
      shortName: `${trimmed.slice(0, 45).trim()}…`,
      oneLineReason: null,
    };
  }

  return { shortName: trimmed, oneLineReason: null };
}

File: ios-client/OutfitSuggestor/Utils/OutfitItemCardText.swift
//
//  OutfitItemCardText.swift
//  OutfitSuggestor
//

import Foundation

struct OutfitItemCardText {
    let shortName: String
    let oneLineReason: String?
}

enum OutfitItemCardTextParser {
    static func parse(_ description: String) -> OutfitItemCardText {
        let trimmed = description.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else {
            return OutfitItemCardText(shortName: "", oneLineReason: nil)
        }

        let separators = [" — ", " - ", ", ", ". "]
        for sep in separators {
            if let range = trimmed.range(of: sep) {
                let idx = trimmed.distance(from: trimmed.startIndex, to: range.lowerBound)
                if idx > 0, idx < 60 {
                    let shortName = String(trimmed[..<range.lowerBound]).trimmingCharacters(in: .whitespacesAndNewlines)
                    let rest = String(trimmed[range.upperBound...]).trimmingCharacters(in: .whitespacesAndNewlines)
                    let reason = rest.isEmpty ? nil : String(rest.prefix(80))
                    return OutfitItemCardText(shortName: shortName, oneLineReason: reason)
                }
            }
        }

        if trimmed.count > 48 {
            let prefix = String(trimmed.prefix(45)).trimmingCharacters(in: .whitespacesAndNewlines)
            return OutfitItemCardText(shortName: "\(prefix)…", oneLineReason: nil)
        }

        return OutfitItemCardText(shortName: trimmed, oneLineReason: nil)
    }
}
